HR_System.py

```diff
@@ -0,0 +1,309 @@
+# AHernandez, 2021-May-25, Added data structure and input functions 
+# AHernandez, 2021-May-25, Added entering new user, exporting to csv, and printing table
+# AHernandez, 2021-May-26, added exporting table to text as a report  
+# AHernandez, 2021-May-27, Added importing file functionality  
+# AHernandez, 2021-May-28, Added Edit employee functionality
+# AHernandez, 2021-May-29, Added Review Report and export
+# ------------------------------------------#
+import time
+import re
+from datetime import datetime 
+import pandas as pd
+import pathlib
+import csv
+from prettytable import PrettyTable
+
+strChoice = '' 
+lstTbl = []
+dictRow = {}
+menuOptions = '\nMenu:\n\t[R] Create Upcoming Review Report\n\t[L] Load Employee File\n\t[S] Save Current List\n\t[C] Current Employees \
+                \n\t[P] Previous Employees\n\t[A] Add Employee\n\t[E] Edit Current Employee\n\t[X] Quit'
+strMenuOption = 'r','l','s','c','p','a','e','x'
+strFileName = 'Employee.csv'
+strRegexSSN = '^(?!000|666)[0-8][0-9]{2}-(?!00)[0-9]{2}-(?!0000)[0-9]{4}$'
+fieldnames = ['EmpId','Name', 'Address', 'SSN', 'DOB', 'jobTitle', 'startDate', 'EndDate']
+ptable = ''
+
+# -- PROCESSING -- #
+class DataProcessor:
+    def append_employee(Name, Address, SSN ,DOB, jobTitle, startDate,EndDate, empId):
+        dictRow[empId] = [Name, Address, SSN ,DOB, jobTitle, startDate, EndDate]
+        return dictRow
+    
+    def find_current_employee(dictRow):
+        try:
+            for key, val in dictRow.items():
+                table = PrettyTable(fieldnames)
+                for key, val in dictRow.items():
+                    if datetime.now() < datetime.strptime(val[6], '%Y-%m-%d'):
+                        table.add_row([key, *val])
+            return table 
+        except UnboundLocalError:
+            print('The table is empty.  There is nothing to print.')   
+            
+    def find_previous_employee(dictRow):
+        try:
+            for key, val in dictRow.items():
+                table = PrettyTable(fieldnames)
+                for key, val in dictRow.items():
+                    if datetime.now() > datetime.strptime(val[6], '%Y-%m-%d'):
+                        table.add_row([key, *val])
+            return table 
+        except UnboundLocalError:
+            print('The table is empty.  There is nothing to print.')
+    
+    def find_emp_record(empOpt,dictRow):
+        try:
+            for key in dictRow.keys():
+                if empOpt == int(key):
+                    empId = int(key)
+            return empId
+        except UnboundLocalError:
+            print(f'Employee Id {empOpt} does not exist')
+            
+    def update_emp_record(Name, Address, SSN ,DOB, jobTitle, startDate,EndDate, empId,dictRow):
+        for key in dictRow.keys():
+            if int(key) == empId:
+                dictRow[key] = [Name, Address, SSN ,DOB, jobTitle, startDate, EndDate]
```

Comment:
You can simplify this to `dictRow[empId]=[Name, Address, SSN ,DOB, jobTitle, startDate, EndDate]`